protocol_rfcs/catalog-owned.md

```diff
@@ -0,0 +1,489 @@
+catalog.
+
+This component then has several options for proposing, ratifying, and publishing the commit,
+detailed below.
+
+- Option 1: Write the actions (likely client-side) to an [staged commit file](#staged-commit) in the
+  `_delta_log/_staged_commits` directory and then ratify the staged commit by atomically recording
+  (likely server-side) (in persistent storage of some kind) that the file corresponds to version
+  `v`.
+- Option 2: Treat this as an [inline commit](#inline-commit) (i.e. likely that the client-side
+  component sends the contents to the server-side component) and atomically record the content of
+  the commit as version `v` of the table.
+- Option 3: Propose, ratify, and publish all-in-one by atomically writing a [published commit file](#published-commit)
+  in the `_delta_log` directory. Note that this commit will be considered to have succeeded as soon
+  as the file becomes visible in the filesystem, regardless of when or whether the catalog is made
+  aware of the successful publish. The catalog does not need to track these files.
+
+A catalog must not ratify version `v` until it has ratified version `v - 1`, and it must ratify
+version `v` at most once.
+
+The catalog must track both flavors of ratified commits (staged or inline) and make them available
+to readers until they are [published](#publishing-commits).
+
+Delta clients are encouraged to establish an API contract where the catalog provides the latest
+ratified commit information whenever a commit fails due to version conflict.
+
+## Getting Ratified Commits from the Catalog
+
+Even after a commit is ratified, it is not discoverable through filesystem operations until it is
+[published](#publishing-commits).
+
+The catalog is responsible to implement an API (defined by the Delta client) that Delta clients can
+use to retrieve the latest ratified commit version (authoritative), as well as the set of ratified
+commits the catalog is still tracking for the table. If some commits needed to complete the snapshot
+are not tracked by the catalog, as they are already published, Delta clients can issue a filesystem
+LIST operation to retrieve them.
+
+Delta clients must establish an API contract where the catalog provides ratified commit information
+as part of the standard table resolution process performed at query planning time.
+
+## Publishing Commits
+
+Publishing is the process of copying the ratified commit with version `<v>` to
+`_delta_log/<v>.json`. The ratified commit may be a staged commit located in
+`_delta_log/_staged_commits/<v>.<uuid>.json`, or it may be an inline commit whose content the
+catalog tracks itself. Because the content of a ratified commit is immutable, it does not matter
+whether the client-side, server-side, or both catalog components initiate publishing.
+
+Implementations are strongly encouraged to publish commits promptly. This reduces the number of
```

Comment:
This is not necessary. In theory a catalog could be fine tracking many commits, and clients could benefit from getting those commits from the catalog instead of from the file system. E.g. if checkpoint interval is 2500 commits, then it might be beneficial if the catalog serves up 2500 commits.

I would suggest simply making the choice of when to publish the joint responsibility of the client and the catalog.

Reply:
I'll just delete this sentence then?